### ISSUE TITLE
[Bug]: Olive downloads a non-existent .nupkg package for Microsoft.ML.ONNXRuntime

### ISSUE THREAD

**Issue by jacob-mink** [microsoft/Olive#522]
### What happened?

I'm not sure this will be able to reproduce, as it's totally dependent on the status of the ort-nightly Azure DevOps builds.

I noticed that VS refused to load up the Microsoft.ML.ONNXRuntime .nupkg from the model zip folder after running the workflow to package up openai/whisper-tiny.

Steps to reproduce:
1. pip install ort-nightly at some 1.16.0 dev version for which there is no corresponding Microsoft.ML.ONNXRuntime nuget package available at https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/NuGet/Microsoft.ML.OnnxRuntime/versions/ (at the time of writing, 8/28/2023's ort-nightly python package had no corresponding Microsoft.ML.ONNXRuntime nuget package)
2. Run the workflow to generate the openai/whisper-tiny ONNX model
3. Unzip the resulting folder and either a) unzip the .nupkg with a tool like 7zip or b) set that folder as a local nuget repository in visual studio

After step 3, you should either see a failure to unzip or a failure to show the package in the NuGet package manager.

Hints I think I found.
1) The file is created no matter the result of pulling the file from the web
2) I think the Azure DevOps naming convention for the ONNXRuntime nuget package is different than the naming for the python ort-nightly package
3) For Microsoft.ML.ONNXRuntime at some nightly version to work, you'd need (at least) the corresponding Microsoft.ML.ONNXRuntime.Managed nightly version, as well.

This is, obviously, non-blocking as the solution is to go back in time to an ort-nightly for which a corresponding Microsoft.ML.ONNXRuntime already exists, but I'm sure it'll trip someone up who won't get as lucky as I did doing version inspection. I wonder if the user experience should be failing the build if they're using an ort-nightly that does not have corresponding runtime support... Or if it's even feasible to find the NuGet packages given that the naming convention is so off!

**tl;dr the downloaded nuget package is not real if the package does not exist on Azure DevOps and/or if the naming convention does not match**

### Version?

84ac609

**Comment by xiaoyu-work** [comment]
Thanks for your feedback. Looking at this.

**Comment by guotuofeng** [comment]
@xiaoyu-work, are there any updates?

**Comment by xiaoyu-work** [comment]
I temporally removed ort-nightly downloading logic for nuget package as there is no directly download link for it: https://github.com/microsoft/Olive/pull/588. Will keep tracking this with Azure DevOps team if we can download it programmatically.

**Comment by xiaoyu-work** [comment]
The code had been merged. Now no wrong ort-nightly package will be downloaded.
